### PR TITLE
PRODENG-2281 fix mcrconfig value parsing

### PR DESCRIPTION
**Pull request by james-nesbitt**
# Description

- existing parsing was not properly building cluster config
- switch to nested attribute instead of nested blocks

Also

- schema transform was supposed to return errors but it never
  did, so that is removed in favour of adding diags

## Issue

https://mirantis.jira.com/browse/PRODENG-2281

### CODE REVIEW

**Review by cranzy — APPROVED** [review]
Good catch

**Comment by james-nesbitt** [comment]
I will wait for @nekwar to check it out before merging.

**Comment by james-nesbitt** [comment]
I rewrote this to use nested attributes at the request of @nekwar

File: README.md [hunk 1]
```diff
@@ -84,10 +84,16 @@ No requirements.
...
 
+No modules.
+
 ## Resources
 
+No resources.
+
 ## Inputs
 
 No inputs.
```
Comment (james-nesbitt) [inline README.md:99]: README.md changes are automated

**Review by nekwar — APPROVED** [review]

**Pull request merged**